### ISSUE TITLE
sourcemap inline doesn't work when using API

### ISSUE THREAD

**Issue by electerious** [rollup/rollup#3913]
- Rollup Version: 2.35.1
- Operating System (or Browser): macOS 10.15.7
- Node Version (if applicable): v14.5.0

```js
const bundle = await rollup.rollup({
	input: filePath,
})
const { output } = await bundle.generate({
	format: 'iife',
	sourcemap: 'inline'
})

console.log(output[0].code) // Expect sourcemap to be part of the code (= inline)
console.log(output[0].map) // Sourcemap is only available via .map
```

See https://repl.it/@electerious/rollup-repro#index.js (run with `node index.js`)

### Expected Behavior

I'm using the API and passing the `sourcemap: 'inline'` option to Rollup (output options). I've expected that the sourcemap is inline and therefore a part of `output[0].code`.

### Actual Behavior

The sourcemap is only available via `output[0].map`, but not part of `output[0].code`.

That might be the correct behaviour when using the API, but there's nothing in the docs telling me that 'inline' won't work when using the API.

**Comment by shellscape** [comment]
Thanks for opening an issue. Citing the issue template:

>   Issues without minimal reproductions will be closed! Please provide a link to one by:
>   1. Using the REPL at https://rollupjs.org/repl/, or
>   2. Using the REPL.it reproduction template at https://repl.it/@rollup/rollup-repro
>      (allows full use of all rollup options and plugins), or
>   3. Provide a minimal repository link (Read https://git.io/fNzHA for instructions).
>      These may take more time to triage than the other options.

Please add a reproduction and we'll be happy to triage further.

**Comment by electerious** [comment]
Reproduction: https://repl.it/@electerious/rollup-repro#index.js

**Comment by shellscape** [comment]
thank you

**Comment by lukastaegert** [comment]
Finally: Fix at #4605
